smarts/core/utils/traffic_history_serivce.py

```diff
@@ -177,5 +192,8 @@ def fetch_agent_missions(history_file_path):
                         goal=scenario.EndlessGoal(),
                         start_time=float(t),
                     )
+        history_mission[history_file_path] = vehicle_missions
+        with open(history_mission_filepath, "wb") as f:
+            pickle.dump(history_mission, f)
```

Comment:
since each `agent_missions` object could be big, it might be helpful to use history_mission as a set to store paths to `agent_missions` pickle files instead of storing all `agent_missions` in 1 file. 

So this would be the structure of util folder
 utils 
      - temp_vehicle_mission.pkl 
      - tmp_agent_mission_folder
        -  agent_mission_for_scenario_A 
        -  agent_mission_for_scenario_B
So temp_vehicle_mission.pkl would store `("tmp_agent_mission_folder/agent_mission_for_scenario_A", "tmp_agent_mission_folder/agent_mission_for_scenario_B")`. 

However, I am not sure if `agent_missions` would be that big for this to be important.